src/main/java/edu/kis/powp/jobs2d/command/manager/DriverCommandManager.java

```diff
@@ -47,6 +48,24 @@ public void execute(Job2dDriver driver) {
 				return driverCommands.iterator();
 			}
 
+			@Override
+			public ICompoundCommand clone() {
+				ICompoundCommand command = null;
+				try {
+					command = (ICompoundCommand) super.clone();
+				} catch (CloneNotSupportedException e) {
+					e.printStackTrace();
+				}
+				
+				Iterator<DriverCommand> source = this.iterator();
+				//List type is irrelevant for this purpouse
+				List<DriverCommand> target = new ArrayList<DriverCommand>();
+				source.forEachRemaining(val -> target.add(val.clone()));
+				driverCommands = target;
```

Comment:
What is the purpose of those lines? Why cloning should affect the cloned object?
Maybe instead of hacking this anonymous class instance provide an ICompoundCommand implementation or try to implement a default clone in the interface (with actual CloneNotSupportedException exception handling)

Reply:
I see. Will take a look on that. I'm not too fluent in java to be honest.

Providing an implementation for ICompoundCommand will make this only applicable to this one specific implementation. Will that be ok as solution for this task?

What do you mean by 'default clone in the interface'? Is there a possibility to implement something in interface? I was thinking at the beginning on changing it simply to the abstract class instead of interface, but in the end decided to not change current code too much.

Reply:
Providing at least one working implementation would be ok. 

And yes, java enables providing a default implementation in interfaces (default keyword). It only uses other declared operations, because interfaces cannot have fields (of course :) ). So e.g., you can implement clone method and use iterator() method in it.

Reply:
Updated solution with concrete implementation of CompoundCommand.